mu/interface/main.py

```diff
@@ -1357,6 +1360,70 @@ def hide_device_selector(self):
         """
         self.status_bar.device_selector.setHidden(True)
 
+    def upload_to_python_anywhere(
+        self, instance, username, token, app_name, files
+    ):
+        """
+        Show a progress dialog as the files are uploaded to PythonAnywhere.
+        """
+        self.progress = QProgressDialog(
+            _(
+                "Uploading to PythonAnywhere.\n\nThis may take some time "
+                "and pause at the end (as the website is started)."
+            ),
+            None,
+            0,
+            len(files),
+            self,
+        )
+        self.progress.setModal(True)
+        self.progress.setAutoClose(True)
+        self.progress.show()
+        self.upload_thread = QThread()
+        self.worker = PythonAnywhereWorker(
+            instance, username, token, files, app_name, self.progress
+        )
+        self.worker.moveToThread(self.upload_thread)
+        self.upload_thread.started.connect(self.worker.run)
+        self.worker.finished.connect(self.upload_thread.quit)
+        self.worker.finished.connect(self.worker.deleteLater)
+        self.worker.finished.connect(self.handle_python_anywhere_complete)
+        self.upload_thread.finished.connect(self.upload_thread.deleteLater)
+        self.worker.error.connect(self.handle_python_anywhere_error)
+        self.worker.error.connect(self.upload_thread.quit)
+        self.worker.error.connect(self.worker.deleteLater)
+        self.upload_thread.start()
+
+    def handle_python_anywhere_complete(self, domain):
+        """
+        Displays a confirmation that all the API calls completed OK and
+        provides a link to the user's website.
+        """
+        message = _(
+            "<p>Your website was successfully deployed.</p>"
+            "<p>You can see it here:</p>"
+            "<a href='https://{domain}'>https://{domain}</a>"
+        ).format(domain=domain)
+        self.show_message(
+            message,
+            icon="Information",
+        )
+
+    def handle_python_anywhere_error(self, error_message):
```

Comment:
Hmm... this is a tricky one. 

When I was writing the code for this I thought along the following lines: the error will be contained in Mu's logs, and if it isn't network related (and the HTTP related errors are explicit in saying they're network related), the user can just get in touch with us per the usual channels. 

Also, given the addition of networks, the range of errors could be very large and so I went for a catch all approach.

I've updated the text in the error reporting dialog to include a link to submit a bug here on GitHub (the simplest solution). I could make the URL for the link more complicated and use our existing codewith.mu related mechanism for bug reporting, but this will require a non-trivial amount of work to achieve. In any case, it now looks like this:

![error](https://user-images.githubusercontent.com/37602/174619157-f7e95759-ad26-44bf-9cdf-62829e2900f9.png)

Reply:
Agree, I think the new message is good enough, thanks!